otherlibs/win32unix/utimes.c

```diff
@@ -0,0 +1,82 @@
+/**************************************************************************/
+/*                                                                        */
+/*                                 OCaml                                  */
+/*                                                                        */
+/*             Xavier Leroy, projet Cristal, INRIA Rocquencourt           */
+/*                                                                        */
+/*   Copyright 2017 Institut National de Recherche en Informatique et     */
+/*     en Automatique.                                                    */
+/*                                                                        */
+/*   All rights reserved.  This file is distributed under the terms of    */
+/*   the GNU Lesser General Public License version 2.1, with the          */
+/*   special exception on linking described in the file LICENSE.          */
+/*                                                                        */
+/**************************************************************************/
+
+#define CAML_INTERNALS
+
+#include <caml/fail.h>
+#include <caml/mlvalues.h>
+#include <caml/memory.h>
+#include <caml/signals.h>
+#include <caml/osdeps.h>
+#include "unixsupport.h"
+
+#include <windows.h>
+
+static void convert_time(double unixTime, FILETIME* ft)
+{
+  ULARGE_INTEGER u;
+  /* There are 11644473600 seconds between 1 January 1601 (the NT Epoch) and 1
+   * January 1970 (the Unix Epoch). FILETIME is measured in 100ns ticks.
+   */
+  u.QuadPart = (ULONGLONG)(unixTime * 10000000.0) + INT64_LITERAL(116444736000000000U);
+  ft->dwLowDateTime = u.LowPart;
+  ft->dwHighDateTime = u.HighPart;
+}
+
+CAMLprim value unix_utimes(value path, value atime, value mtime)
+{
+  CAMLparam3(path, atime, mtime);
+  WCHAR *wpath;
+  HANDLE hFile;
+  FILETIME lastAccessTime, lastModificationTime;
+  SYSTEMTIME systemTime;
+  double at, mt;
+  BOOL res;
+
+  caml_unix_check_path(path, "utimes");
+  at = Double_val(atime);
+  mt = Double_val(mtime);
+  wpath = caml_stat_strdup_to_utf16(String_val(path));
+  caml_enter_blocking_section();
+  hFile = CreateFile(wpath,
+                     GENERIC_WRITE,
```

Comment:
I think `FILE_WRITE_ATTRIBUTES` is sufficient here.

Reply:
Indeed, fixed.